### PR TITLE
Change telegram library to pytelegrambotapi

### PR DESCRIPTION
**Pull request by rgstephens**
**Proposed changes**:
Change Telegram library to address #6814.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/channels/telegram.py [hunk 1]
```diff
@@ -202,11 +209,12 @@ async def set_webhook(_: Request) -> HTTPResponse:
         async def message(request: Request) -> Any:
             if request.method == "POST":
 
-                if not out_channel.get_me()["username"] == self.verify:
+                request_dict = request.json
+                update = Update.de_json(request_dict)
```
Comment (erohmensing) [inline rasa/core/channels/telegram.py:213]: what happened to the `out_channel` argument?

File: rasa/core/channels/telegram.py [hunk 2]
```diff
@@ -33,11 +34,13 @@ async def send_text_message(
...
+            logger.debug(f"message_part: {message_part}")
             self.send_message(recipient_id, message_part)
 
     async def send_image_url(
         self, recipient_id: Text, image: Text, **kwargs: Any
     ) -> None:
+        logger.debug(f"image: {image}")
         self.send_photo(recipient_id, image)
 
     async def send_text_with_buttons(
```
Comment (erohmensing) [inline rasa/core/channels/telegram.py:46]: Re: line 51, it seems like vertical is the default now (using moodbot0? I would double check that passing these `button_type` args works as expected

File: rasa/core/channels/telegram.py [hunk 3]
```diff
@@ -59,38 +62,42 @@ async def send_text_with_buttons(
...
 
         elif button_type == "vertical":
-            button_list = [
-                [InlineKeyboardButton(s["title"], callback_data=s["payload"])]
-                for s in buttons
-            ]
-            reply_markup = InlineKeyboardMarkup(button_list)
+            reply_markup = ReplyKeyboardMarkup(
+                resize_keyboard=True, one_time_keyboard=True
+            )
```
Comment (erohmensing) [inline rasa/core/channels/telegram.py:77]: I don't think vertical should be ReplyKeyboardMarkup, but rather just an inline one with one per line?

File: rasa/core/channels/telegram.py [hunk 4]
```diff
@@ -59,38 +62,42 @@ async def send_text_with_buttons(
...
+            for idx, button in enumerate(buttons):
+                if isinstance(button, list):
+                    reply_markup.add(KeyboardButton(s["title"]) for s in button)
```
Comment (erohmensing) [inline rasa/core/channels/telegram.py:80]: Why would `button` (in a buttons list) also be a list?

**Review by erohmensing — APPROVED** [review]
A few comments to address before merging, but seems to work as expected

File: changelog/6814.improvement.md [hunk 5]
```diff
@@ -0,0 +1 @@
+Replace use of `python-telegram-bot` package with `pyTelegramBotAPI`
```
Comment (erohmensing) [inline changelog/6814.improvement.md:1]:
```suggestion
Replace use of ``python-telegram-bot`` package with ``pyTelegramBotAPI``
```
RST -- make this change to the docs too

File: docs/user-guide/connectors/telegram.rst [hunk 6]
```diff
@@ -44,3 +44,43 @@ you need to supply a ``credentials.yml`` with the following content:
      access_token: "490161424:AAGlRxinBRtKGb21_rlOEMtDFZMXBl6EC0o"
      verify: "your_bot"
      webhook_url: "https://your_url.com/webhooks/telegram/webhook"
+
+In addition to standard `text:` utterances, this channel also supports the following components from the `Telegram API <https://core.telegram.org/bots/api/>`_
```
Comment (erohmensing) [inline docs/user-guide/connectors/telegram.rst:48]: https://core.telegram.org/bots/api/#message

File: rasa/core/channels/telegram.py [hunk 7]
```diff
@@ -19,9 +19,10 @@
 logger = logging.getLogger(__name__)
 
 
-class TelegramOutput(Bot, OutputChannel):
+class TelegramOutput(TeleBot, OutputChannel):
     """Output channel for Telegram"""
 
+    # skipcq: PYL-W0236
```
[the remark this answers is not in the document]
Comment (rgstephens) [reply rasa/core/channels/telegram.py:25]: Looks like this was added by Tom last month and is related to [this](https://deepsource.io/gh/RasaHQ/rasa/issue/PYL-W0236/occurrences)

File: rasa/core/channels/telegram.py [hunk 8]
```diff
@@ -33,11 +34,13 @@ async def send_text_message(
         self, recipient_id: Text, text: Text, **kwargs: Any
     ) -> None:
         for message_part in text.strip().split("\n\n"):
+            logger.debug(f"message_part: {message_part}")
```
[the remark this answers is not in the document]
Comment (rgstephens) [reply rasa/core/channels/telegram.py:37]: done, thanks

File: rasa/core/channels/telegram.py [hunk 9]
```diff
@@ -59,38 +62,42 @@ async def send_text_with_buttons(
...
         elif button_type == "reply":
-            button_list = []
-            for bttn in buttons:
-                if isinstance(bttn, list):
-                    button_list.append([KeyboardButton(s["title"]) for s in bttn])
-                else:
-                    button_list.append([KeyboardButton(bttn["title"])])
             reply_markup = ReplyKeyboardMarkup(
-                button_list, resize_keyboard=True, one_time_keyboard=True
+                resize_keyboard=False, one_time_keyboard=True
```
[the remark this answers is not in the document]
Comment (rgstephens) [reply rasa/core/channels/telegram.py:86]: That should not have been changed. Thanks.

File: rasa/core/channels/telegram.py (hunk 1 above)
Comment (rgstephens) [reply rasa/core/channels/telegram.py:213]: Not sure how this worked before but `get_me` returns a user object.

File: docs/user-guide/connectors/telegram.rst (hunk 6 above)
Comment (rgstephens) [reply docs/user-guide/connectors/telegram.rst:48]: updated

File: rasa/core/channels/telegram.py (hunk 3 above)
Comment (rgstephens) [reply rasa/core/channels/telegram.py:77]: Yes

File: rasa/core/channels/telegram.py (hunk 4 above)
Comment (rgstephens) [reply rasa/core/channels/telegram.py:80]: fixed

File: rasa/core/channels/telegram.py (hunk 2 above)
Comment (rgstephens) [reply rasa/core/channels/telegram.py:46]: fixed

**Comment by rgstephens** [comment]
@erohmensing I made changes and did full testing of the buttons based on your comments (thanks) but the test for the `setWebhook` endpoint is now failing. The old library has an [alias](https://github.com/python-telegram-bot/python-telegram-bot/blob/264b2c9c72691c5937b80e84e061c52dd2d8861a/telegram/bot.py#L4429) for the `setWebhook` function but it seems to me that the test shouldn't care that the class alias is missing as long as the `setWebhook` endpoint is supported and there's been no change to the blueprint for `setWebhook`.  I'm thinking the test needs to be changed but would like your thoughts on this.  -Thanks.

File: docs/user-guide/connectors/telegram.rst [hunk 10]
```diff
@@ -44,3 +44,43 @@ you need to supply a ``credentials.yml`` with the following content:
      access_token: "490161424:AAGlRxinBRtKGb21_rlOEMtDFZMXBl6EC0o"
      verify: "your_bot"
      webhook_url: "https://your_url.com/webhooks/telegram/webhook"
+
+In addition to standard ``text:`` utterances, this channel also supports the following components from the `Telegram API <https://core.telegram.org/bots/api/#message>`_
+
+- button
```
Comment (erohmensing) [inline docs/user-guide/connectors/telegram.rst:50]:
```suggestion
- ``button`` arguments:
```

File: docs/user-guide/connectors/telegram.rst [hunk 11]
```diff
@@ -44,3 +44,43 @@ you need to supply a ``credentials.yml`` with the following content:
...
+  - button_type: inline | vertical | reply
+- custom payloads:
```
Comment (erohmensing) [inline docs/user-guide/connectors/telegram.rst:52]:
```suggestion
- ``custom`` arguments: 
```

**Review by erohmensing — APPROVED** [review]

**Pull request merged**